retry/lib/retry.dart

```diff
@@ -118,24 +118,38 @@ final class RetryOptions {
   /// If no [retryIf] function is given this will retry any for any [Exception]
   /// thrown. To retry on an [Error], the error must be caught and _rethrown_
   /// as an [Exception].
+  ///
+  /// The [onRetry] `Function` below stands for one of two types:
+  /// - (Exception) -> FutureOr<void>
+  /// - (Exception, StackTrace) -> FutureOr<void>
   Future<T> retry<T>(
     FutureOr<T> Function() fn, {
     FutureOr<bool> Function(Exception)? retryIf,
-    FutureOr<void> Function(Exception)? onRetry,
+    Function? onRetry,
```

Comment:
I wonder if there is an annotation or comment that'll allow us to hide the parameter from auto-completion :D

Reply:
```
@Deprecated('Use `onRetryFailure` instead of `onRetry`')
FutureOr<void> Function(Exception)? onRetry,
FutureOr<void> Function(Exception, StackTrace)? onRetryFailure,
```
Thank for review. 

Do you think 'onRetryFailure' would be a good name for this?

Reply:
Hmm, isn't logging the only legitimate use-case for `onRetry`, hence, `logRetry` might be a good name, or `retryLog` or???

We could also all it `whenRetry`.

Reply:
There are some names here:

* (i) `onFailure`.  regardless of whether Exception satisfies `retryIf`, it will be called.
* (ii) `onError`. any error on retry, it will be called.

Which one do you think is better?